src/components/NewCommentForm.tsx

```diff
@@ -75,25 +204,40 @@ export const NewCommentForm: React.FC = () => {
             id="comment-body"
             name="body"
             placeholder="Type comment here"
-            className="textarea is-danger"
+            className={classNames('textarea', {
+              'is-danger': !hasComment,
+            })}
+            value={message}
+            onChange={handleCommentChange}
           />
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Enter some text
-        </p>
+        {!hasComment && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            {ErrorForm.COMMENT}
+          </p>
+        )}
       </div>
 
       <div className="field is-grouped">
         <div className="control">
-          <button type="submit" className="button is-link is-loading">
+          <button
+            type="submit"
+            className={classNames('button', 'is-link', {
+              'is-loading': isCommentAdd,
+            })}
+            onClick={handleFormSubmit}
+          >
```

Comment:
You are passing handleFormSubmit to the onClick event of the submit button. You should instead just rely on the onSubmit event of the form. Remove onClick from the submit button.